wikiclass/feature_lists/fawiki.py

```diff
@@ -0,0 +1,99 @@
+"""
+Persian Wikipedia
++++++++++++++++++
+"""
+
+from revscoring.features import wikitext
+from revscoring.features.meta import aggregators
+from revscoring.features.modifiers import max, sub, log
+from revscoring.features.wikitext.datasources import Revision
+from revscoring.datasources.meta import mappers, filters
+
+from . import wikipedia
+
+# Templates
+infobox_templates = wikitext.revision.template_names_matching(
+    r"infobox|جعبه", name="fawiki.revision.infobox_templates")
+CN_TEMPLATES = [
+    r"Citation[_ ]needed",
+    r"Cn",
+    r"Fact",
+    r"مدرک"
+]
+cn_templates = wikitext.revision.template_names_matching(
+    "|".join(CN_TEMPLATES), name="fawiki.revision.cn_templates")
+who_templates = wikitext.revision.template_names_matching(
+    "Who|چه کسی|چه‌کسی", name="fawiki.revision.who_templates")
+main_article_templates = wikitext.revision.template_names_matching(
+    "Main|اصلی", name="fawiki.main_article_templates")
+cite_templates = wikitext.revision.template_names_matching(
+    r"cite|یادکرد", name="fawiki.revision.cite_templates")
+proportion_of_templated_references = \
+    cite_templates / max(wikitext.revision.ref_tags, 1)
+non_templated_references = max(wikitext.revision.ref_tags - cite_templates, 0)
+non_cite_templates = sub(
+    wikitext.revision.templates, cite_templates,
+    name="fawiki.revision.non_cite_templates"
+)
+
+# Links
+category_links = wikitext.revision.wikilink_titles_matching(
+    r"Category|رده\:", name="fawiki.revision.category_links")
+image_links = wikitext.revision.wikilink_titles_matching(
+    r"File|Image|پرونده|تصویر\:", name="fawiki.revision.image_links")
+
+# References
+revision = Revision(
+    "fawiki.revision.revision",
+    wikitext.revision.datasources,
+)
```

Comment:
What's going on here?

Reply:
This comes from the enwiki file. I don't know to be exact :D